DBTools/python/getSiblings.py

```diff
@@ -0,0 +1,331 @@
+        else:
+            self.dictName = 'default.json'
+
+        if not self.dictName.endswith('.json'):
+            self.dictName += '.json'
+
+    #get list of files for datasets
+    def setFileLists(self, listType=''):
+        if args.prod == 'allLocal':
+            if listType == 'input' or listType == '': 
+                self.inputFileList = self.getLocalFileList(self.dataset_in)
+            if listType == 'sibling' or listType == '': 
+                self.siblingFileList = self.getLocalFileList(self.dataset_sib)
+        elif args.prod == 'local':
+            if listType == 'input' or listType == '': 
+                self.inputFileList = self.getLocalFileList(self.dataset_in)
+            if listType == 'sibling' or listType == '': 
+                self.siblingFileList = self.getFileList(self.dataset_sib, self.dbsapi_out)
+        else:
+            if listType == 'input' or listType == '': 
+                self.inputFileList = self.getFileList(self.dataset_in, self.dbsapi_in)
+            if listType == 'sibling' or listType == '': 
+                self.siblingFileList = self.getFileList(self.dataset_sib, self.dbsapi_out)
+
+    #get the dbs api for a given dataset
+    def setAPIs(self):
+        try:
+            from dbs.apis.dbsClient import DbsApi
+        except ImportError:
+            print("getSiblings() relies on CRAB. Please set up the environment for CRAB before using.")
+            sys.exit (1)
+        if args.prod == 'local':
+            self.dbsapi_out = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader')
+        elif args.prod == 'global':
+            self.dbsapi_in = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader')
+            self.dbsapi_out = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader')
+        elif args.prod == 'user':
+            self.dbsapi_in = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader')
+            self.dbsapi_out = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/global/DBSReader')
+        elif args.prod == 'allUser':
+            self.dbsapi_in = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader')
+            self.dbsapi_out = DbsApi (url = 'https://cmsweb.cern.ch/dbs/prod/phys03/DBSReader')
+
+    #main function for getting siblings using either lumi matching or DBS
+    def getSiblings(self):
+        if args.prod == 'local' or args.prod == 'allLocal' or args.lumiMatching:
+            self.getLumiMatching(self.dictName, args.prod, args.inputJSON, args.siblingJSON)
+        else:
+            self.getDBSSiblings(self.dictName)
+
+    #function to get siblings from DBS
+    def getDBSSiblings(self, output_json):
+        print("getting dbs siblings")
+
+        file_dict = {}
+
+        print("Output JSON file will be called {0}".format(self.dictName))
+        #print('Input dataset: {}'.format(self.dataset_in))
```

Comment:
Maybe remove this line or add a debug/verbose option to print it if it can be useful for debugging

Reply:
removed